### PR TITLE
feat: update hermes to support publisher stake caps

### PR DESCRIPTION
**Pull request by guibescos**
This PR adds a new endpoint `/api/get_publisher_stake_caps_update_data` for consumption of publisher caps messages

### CODE REVIEW

File: apps/hermes/server/src/api.rs [hunk 1]
```diff
@@ -150,6 +150,10 @@ where
         .route("/api/get_price_feed", get(rest::get_price_feed))
         .route("/api/get_vaa", get(rest::get_vaa))
         .route("/api/get_vaa_ccip", get(rest::get_vaa_ccip))
+        .route(
```
Comment (ali-bahjati) [inline apps/hermes/server/src/api.rs:153]: we have deprecated the old api endpoints and don't add things to them. could you please move it to `v2` and use something like `v2/updates/publisher_stake_caps/latest`?

File: apps/hermes/server/src/api/types.rs [hunk 2]
```diff
@@ -271,9 +273,28 @@ impl From<PriceFeedUpdate> for ParsedPriceUpdate {
...
+    pub publisher: Pubkey,
+    pub cap:       u64,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize, ToSchema)]
+pub struct GetPublisherStakeCapsUpdateDataResponse {
+    pub binary: BinaryUpdate,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub parsed: Option<Vec<ParsedPublisherStakeCapsUpdate>>,
+}
```
Comment (ali-bahjati) [inline apps/hermes/server/src/api/types.rs:293]: This is only used in a single api right? please move it there.

File: apps/hermes/server/src/state/aggregate.rs [hunk 3]
```diff
@@ -403,6 +417,41 @@ where
...
+            .collect::<Result<Vec<_>>>()?;
+
+        let update_data = construct_update_data(messages.into_iter().map(|m| m.into()).collect())?;
+        Ok(PublisherStakeCapsWithUpdateData {
+            publisher_stake_caps,
+            update_data,
+        })
+    }
+
     async fn get_price_feed_ids(&self) -> HashSet<PriceIdentifier> {
```
Comment (ali-bahjati) [inline apps/hermes/server/src/state/aggregate.rs:455]: Please also fix this method.

**Review by ali-bahjati — CHANGES_REQUESTED** [review]
The code looks good, I left some comments about some structural changes in the code. I think that publishers cap is not very compatible with the terminologies in Hermes and we need a larger refactor (probably later) to handle them.

File: apps/hermes/server/src/api.rs (hunk 1 above)
Comment (guibescos) [reply apps/hermes/server/src/api.rs:153]: done

File: apps/hermes/server/src/api/types.rs (hunk 2 above)
Comment (guibescos) [reply apps/hermes/server/src/api/types.rs:293]: done

**Review by ali-bahjati — APPROVED** [review]

**Pull request merged**